### ISSUE TITLE
debug-info: cannot get lazy loading to work

### ISSUE THREAD

**Issue by x1B** [LaxarJS/laxar-loader#32]
I am trying to lazy-load the debug-info bundle from the [LaxarJS v2 shop demo](https://github.com/LaxarJS/shop-demo). Apparently, the debug contents get inlined into the main bundle even when loading with `?lazy` and setting an `output.chunkFilename`.

Marking as `bug` for now (unless this turns out to be a usage error).

**Comment by jpommerening** [comment]
I believe I have identified the problem!
Our code boils down to:
```
if( /lazy/.test(__resourceQuery) ) {
   require.ensure( … );
} else {
   require( … );
}
```

It seems webpack does not evaluate `RegExp.prototype.test` (grep for `evaluate CallExpression`). I was certain it did at some point? Anyway; it can apparently handle `String.prototype.replace` with regexps, so we could just use it like that:

```
if( __resourceQuery !== __resourceQuery.replace( /lazy/, '$1' ) ) {
```

Definitely worth a source code comment with regards to webpack's parsing.

**Comment by jpommerening** [comment]
Also, a close inspection of the sources revealed we can drop the bundle loader dependency in favor of plain `require.ensure`.

**Comment by jpommerening** [comment]
fixed on `master` & released w/ `v2.0.1`
